packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -690,7 +690,13 @@ class ReactSixteenAdapter extends EnzymeAdapter {
           renderedEl = transformSuspense(renderedEl, renderedEl, { suspenseFallback });
           const { type: Component } = renderedEl;
 
-          const context = getMaskedContext(Component.contextTypes, unmaskedContext);
+          let context;
+          if (Component.contextType) {
+            const Provider = adapter.getProviderFromConsumer(Component.contextType);
```

Comment:
what happens in React when `.contextType` isn't set to a proper `Consumer`?

Reply:
What do you mean with a proper consumer? A consumer without a parent provider? It should display the default provider value.

Reply:
like, for example, what if `Component.contextType` is a random value - some object, or function, or primitive? What exact check does React do?